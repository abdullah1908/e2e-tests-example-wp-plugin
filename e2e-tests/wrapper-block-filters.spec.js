/**
 * WordPress dependencies
 */
import {
	activatePlugin,
	deactivatePlugin,
	createNewPost,
	getEditedPostContent,
	insertBlock,
	enablePageDialogAccept,
} from '@wordpress/e2e-test-utils';
import {
	selectBlockByName,
	selectOption,
} from './helper';

describe( 'wrapper block filters', () => {
	beforeAll( async () => {
		enablePageDialogAccept();
		await activatePlugin( 'e2e-tests-example-wrapper-block-filters' );
	} );

	afterAll( async () => {
		await deactivatePlugin( 'e2e-tests-example-wrapper-block-filters' );
	} );

	beforeEach( async () => {
		await createNewPost();
	} );

	it( 'e2eTestsExample.wrapperBlock.bgColorOptions should add bg options', async () => {
		await insertBlock( 'Wrapper Block' );
		await selectBlockByName( 'e2e-tests-example/wrapper-block' );

		// Additional style option should be available
		expect( await page.$( 'select.components-select-control__input > option[value="green"]' ) ).not.toBeNull();

		// Style option should be applied
		await selectOption( 'Background Color', 'green' );

		// Editor content should match snapshot
		expect( await getEditedPostContent() ).toMatchSnapshot();
	} );
} );
