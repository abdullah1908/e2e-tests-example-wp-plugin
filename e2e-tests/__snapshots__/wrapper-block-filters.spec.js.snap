// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`wrapper block filters e2eTestsExample.wrapperBlock.bgColorOptions should add bg options 1`] = `
"<!-- wp:e2e-tests-example/wrapper-block {\\"bgColor\\":\\"green\\"} -->
<div style=\\"background-color:green\\" class=\\"wp-block-e2e-tests-example-wrapper-block\\"></div>
<!-- /wp:e2e-tests-example/wrapper-block -->"
`;
